src/Microsoft.Health.Fhir.Core/Features/Operations/ConvertData/ContainerRegistryTemplateProvider.cs

```diff
@@ -52,35 +52,27 @@ public class ContainerRegistryTemplateProvider : IConvertDataTemplateProvider, I
         }
 
         /// <summary>
-        /// Fetch template collection from container registry or built-in archive.
+        /// Fetch template collection from container registry following a custom template convert request.
         /// </summary>
         /// <param name="request">The convert data request which contains template reference.</param>
         /// <param name="cancellationToken">Cancellation token to cancel the fetch operation.</param>
         /// <returns>Template collection.</returns>
         public async Task<List<Dictionary<string, Template>>> GetTemplateCollectionAsync(ConvertDataRequest request, CancellationToken cancellationToken)
         {
-            // We have embedded a default template collection in the templatemanagement package.
-            // If the template collection is the default reference, we don't need to retrieve token.
             var accessToken = string.Empty;
```

Comment:
Do we still need this? or we can merge this line just into line 74:

var accessToken = await _cache.GetOrCreateAsync(GetCacheKey(request.RegistryServer), TokenEntryFactory);